src/core/abstract_types.jl

```diff
@@ -18,6 +18,21 @@ function encode_symbol(
     return Symbol("$(IS.strip_module_name(string(U)))_$(T_)" * meta_)
 end
 
+function encode_symbol(
+    ::Type{T},
+    ::Type{U},
+    ::Type{V},
+    meta::String = CONTAINER_KEY_EMPTY_META,
+) where {T <: PSY.Component, U, V}
+    # TODO DT: it would be better to use "__" in case we need to decode and some names have "_"
```

Comment:
Do we want to do this? Or is it guaranteed that `_` won't exist in the fields?

Reply:
There shouldn't be any names with `_` anymore if I am correct.